gokart/mypy.py

```diff
@@ -0,0 +1,369 @@
+"""Plugin that provides support for gokart.TaskOnKart.
+
+This Code reuses the code from mypy.plugins.dataclasses
+https://github.com/python/mypy/blob/0753e2a82dad35034e000609b6e8daa37238bfaa/mypy/plugins/dataclasses.py
+"""
+
+from __future__ import annotations
+
+from typing import Callable, Final, Iterator, Literal
+
+from mypy.expandtype import expand_type, expand_type_by_instance
+from mypy.nodes import (
+    ARG_NAMED,
+    ARG_NAMED_OPT,
+    ARG_POS,
+    Argument,
+    AssignmentStmt,
+    Block,
+    CallExpr,
+    ClassDef,
+    Context,
+    EllipsisExpr,
+    Expression,
+    FuncDef,
+    IfStmt,
+    JsonDict,
+    MemberExpr,
+    NameExpr,
+    PlaceholderNode,
+    RefExpr,
+    Statement,
+    SymbolTableNode,
+    TempNode,
+    TypeInfo,
+    Var,
+)
+from mypy.plugin import (
+    ClassDefContext,
+    Plugin,
+    SemanticAnalyzerPluginInterface,
+)
+from mypy.plugins.common import (
+    add_method_to_class,
+    deserialize_and_fixup_type,
+)
+from mypy.server.trigger import make_wildcard_trigger
+from mypy.state import state
+from mypy.typeops import map_type_from_supertype
+from mypy.types import (
+    AnyType,
+    CallableType,
+    Instance,
+    NoneType,
+    Type,
+    TypeOfAny,
+    get_proper_type,
+)
+from mypy.typevars import fill_typevars
+
+METADATA_TAG: Final[str] = 'task_on_kart'
+
+
+class TaskOnKartPlugin(Plugin):
+    def get_base_class_hook(self, fullname: str) -> Callable[[ClassDefContext], None] | None:
+        if 'gokart.task.luigi.Task' in fullname:
+            # gather attibutes from gokart.TaskOnKart
+            # the transformation does not affect because the class has `__init__` method
+            return self._task_on_kart_class_maker_callback
```

Comment:
This is an entrypoint to transform a class to have a `__init__` method with `luigi. Parameter`s while type-checking for `gokart.TaskOnKart` class